File: 프로그래머스/2021 KAKAO BLIND RECRUITMENT/신규 아이디 추천.py
# 신규 아이디 추천
def solution(new_id):
    answer = ""
    # 1단계
    new_id = new_id.lower()
    # 2단계
    for word in new_id:
        if word.isalpha() or word.isdigit() or word == '-' or word == '_' or word == '.':
            answer += word
    # 3단계
    tmp = ''
    while True:
        tmp = answer.replace('..','.')
        if len(tmp) == len(answer):
            break
        answer = tmp
    # 4단계
    answer = answer.lstrip(".").rstrip(".")
    # 5단계
    if len(answer) == 0:
        answer = 'a'
    # 6단계
    if len(answer) >= 16:
        answer = answer[:15].rstrip('.')
    # 7단계
    if len(answer) <= 2:
        answer = answer + answer[-1]*3 
        answer = answer[:3]
    return answer

File: 프로그래머스/2021 KAKAO BLIND RECRUITMENT/신규 아이디 추천.js
// 신규 아이디 추천
function solution(new_id) {
    let answer = '';
    // 1단계
    answer = new_id.toLowerCase();
    // 2단계
    let special = /[a-zA-Z0-9-_.]/;
    answer = answer.split("").filter(e => special.test(e)).join("");
    // 3단계
    answer = answer.replace(/\.+/g, '.')
    // 4단계
    answer = answer.replace(/^\.|\.$/g, '') 
    // 5단계
    answer = answer.replace(/^$/, 'a')
    // 6단계
    answer = answer.slice(0, 15).replace(/\.$/, '');
    // 7단계
    const l = answer.length
    answer = l > 2 ? answer : answer + answer.charAt(l - 1).repeat(3 - l);
    return answer;
}